GPy/core/gp.py

```diff
@@ -378,27 +373,28 @@ def predict_jacobian(self, Xnew, kern=None, full_cov=True):
             dK_dXnew_full[i] = kern.gradients_X(one, Xnew, self._predictive_variable[[i]])
 
         if full_cov:
-            dK2_dXdX = kern.gradients_XX(one, Xnew, cov=False)
+            dK2_dXdX = kern.gradients_XX(one, Xnew)
         else:
```

Comment:
It has to do with dimension in this sense that here a sample is a vector of the Jacobian (one vector for each dimension Q). 

The bug in the previous version of the code was due to the fact explained in the (deleted) note:

Note: We always return sum in input_dim gradients, as the off-diagonals
 -        in the input_dim are not needed for further calculations.
 -        This is a compromise for increase in speed. Mathematically the jacobian would
 -        have another dimension in Q.

That *missing* dimension Q was producing a covariance function which was zero in the off-diagonal terms relative to the input dimension, which was wrong. This error was propagated in the computation of the Wishart Embedding.